packages/twenty-server/src/engine/metadata-modules/agent/utils/agent-tool-schema.utils.ts

```diff
@@ -0,0 +1,839 @@
+
+    if (field.isNullable) {
+      result = result.nullable();
+    }
+
+    if (isForUpdate) {
+      result = result.optional();
+    }
+
+    return result;
+  };
+
+  switch (field.type) {
+    case FieldMetadataType.UUID:
+      return createBaseField(z.string().uuid(), 'UUID field');
+
+    case FieldMetadataType.TEXT:
+    case FieldMetadataType.RICH_TEXT:
+      return createBaseField(z.string(), 'Text field');
+
+    case FieldMetadataType.DATE_TIME:
+      return createBaseField(z.string().datetime(), 'Date and time field');
+
+    case FieldMetadataType.DATE:
+      return createBaseField(z.string(), 'Date field (YYYY-MM-DD format)');
+
+    case FieldMetadataType.NUMBER:
+    case FieldMetadataType.NUMERIC:
+    case FieldMetadataType.POSITION:
+      return createBaseField(z.number(), 'Number field');
+
+    case FieldMetadataType.BOOLEAN:
+      return createBaseField(z.boolean(), 'Boolean field');
+
+    case FieldMetadataType.SELECT: {
+      const enumValues =
+        field.options?.map((option: { value: string }) => option.value) || [];
+
+      return createBaseField(
+        z.enum(enumValues as [string, ...string[]]),
+        'Selection field',
+      );
+    }
+
+    case FieldMetadataType.MULTI_SELECT: {
+      const multiEnumValues =
+        field.options?.map((option: { value: string }) => option.value) || [];
+
+      return createBaseField(
+        z.array(z.enum(multiEnumValues as [string, ...string[]])),
+        'Multi-selection field',
+      );
+    }
+
+    case FieldMetadataType.RATING: {
+      const enumValues =
+        field.options?.map((option: { value: string }) => option.value) || [];
+
+      return createBaseField(
+        z.enum(enumValues as [string, ...string[]]),
+        'Rating field',
+      );
+    }
+
+    case FieldMetadataType.ARRAY:
+      return createBaseField(z.array(z.string()), 'Array of strings');
+
+    case FieldMetadataType.LINKS:
+      return createBaseField(
+        z.object({
+          primaryLinkLabel: z.string().optional(),
+          primaryLinkUrl: z.string().optional(),
+          secondaryLinks: z
+            .array(
+              z.object({
+                url: z.string().url().optional(),
+                label: z.string().optional(),
+              }),
+            )
+            .optional(),
+        }),
+        'Links field with primary and secondary links',
+      );
+
+    case FieldMetadataType.CURRENCY:
+      return createBaseField(
+        z.object({
+          amountMicros: z.number().optional(),
+          currencyCode: z.string().optional(),
+        }),
+        'Currency field with amountMicros and currencyCode properties',
+      );
+
+    case FieldMetadataType.FULL_NAME:
+      return createBaseField(
+        z.object({
+          firstName: z.string().optional(),
+          lastName: z.string().optional(),
+        }),
+        'Full name field with firstName and lastName properties',
+      );
+
+    case FieldMetadataType.ADDRESS:
```

Comment:
Same feeling as with the tool service, we need to see if we can be smart and find ways to factorise this instead of duplicating so much similar logic. To discuss!

Reply:
We are using class-validator today which has pretty much the same goal.
There are different options: 
- migrating everything to zod
- converting class validator to zod automatically
- using https://ai-sdk.dev/docs/reference/ai-sdk-core/json-schema and https://www.npmjs.com/package/class-validator-jsonschema (my favorite option for now I think)